### PR TITLE
[READY]Minebots 2022

### PR DESCRIPTION
**Pull request by ghost**
# Document the changes in your pull request

This PR aims to make minebots more usefull and more... cool? So, lets go:

# Maintaining mining bots

Now you will need to open a maintance hatch on the mining bot in order to insert something into it, or remove it's KA upgrades. Additionaly, now you are able to fix their burn damage by replacing burnt wires with new ones. Additionaly, now using multitool on a dead mining bot, while having it's maintance hatch open and health 50% or more, will revive the bot.

If i will finish this PR and it would get merged, i probably will make a PR about hacking this dudes and emaging them for evil deeds.

# Equipment

Now you will be able to install on the mining bot a plasmacutter, a medipen(the bot can use it by alt+clicking on a carbon mob), a mining scanner, a bike horn, and a jaunter(usable by a verb).

The armor upgrade now will be compatitable with the AI upgrade, but the AI upgrade will no more give stat boosts.

# Other things

Names: With this PR you will be able to rename mining bots from boring "mining bot" to something badass, only by using a pen on them. Simple!

GPS signal: Mining bots will be able to toggle a GPS signal, so they owners would be able to find them if something will happen.

Withdrawing ores will be made by alt+clicking a bot, rather then attacking it with a mining scanner.

You no more will be able to switch mining behaviour on a bot, if it is sentient.

# Changelog

:cl: SuperSlayer
rscadd: You can now attach to mining bots plasmacutters, medipens, bike horns, mining scanners, jaunters and better KA's.
tweak: Mining bots now need to have their maintance panel opened to be able to be upgraded
tweak: Mining bots are now renameable
tweak: Armor upgrade no more conflicts with the mining bot AI upgrade
rscadd: Mining bots now have a GPS signal
rscadd: Mining bots are now reviveable by pulsing their wires with a multitool
rscadd: Mining bots burn damage can now be healed by re-wiring them
tweak: Tweaked mining bots controls
/:cl:

### CODE REVIEW

**Comment by ShadowDeath6** [comment]
>reverse-insert(i forgot the word for it, sorry)

remove?

**Comment by ghost** [comment]
> remove?

Yes

File: code/modules/mining/minebot.dm [hunk 1]
```diff
@@ -100,21 +157,118 @@
 		adjustBruteLoss(-15)
 		to_chat(user, span_info("You repair some of the armor on [src]."))
 
+/mob/living/simple_animal/hostile/mining_drone/screwdriver_act(mob/living/user, obj/item/I)
+	. = TRUE
+	maintance_hatch_open = -maintance_hatch_open
```
Comment (nmajask) [inline code/modules/mining/minebot.dm]:
```suggestion
	maintance_hatch_open = !maintance_hatch_open
```

File: code/modules/mining/minebot.dm [hunk 2]
```diff
@@ -100,21 +157,118 @@
...
+		if(I.tool_behaviour == TOOL_CROWBAR || istype(I, /obj/item/borg/upgrade/modkit))
+			I.melee_attack_chain(user, stored_gun, params)
+			scanner.forceMove(get_turf(src))
+			cutter.forceMove(get_turf(src))
+			return
+		else if(istype(I, /obj/item/stack/cable_coil) && health < maxHealth)
+			to_chat(user, span_notice("You begin replacing broken wires on [src]..."))
+			repair_burn(I, user)	
+			return	
+		else if(istype(I, /obj/item/gun/energy/plasmacutter))
```
Comment (nmajask) [inline code/modules/mining/minebot.dm:187]: Doesn't work as you try to weld the bot with the plasma cutter, could fix this by making the bot only weldable if the hatch is closed.

File: code/modules/mining/minebot.dm [hunk 3]
```diff
@@ -37,19 +40,55 @@
 	del_on_death = TRUE
 	var/mode = MINEDRONE_COLLECT
 	var/light_on = 0
-	var/obj/item/gun/energy/kinetic_accelerator/minebot/stored_gun
+	var/maintance_hatch_open = FALSE
```
Comment (nmajask) [inline code/modules/mining/minebot.dm:43]: Should add an inspect text if the hatch is open.

File: code/modules/mining/minebot.dm [hunk 4]
```diff
@@ -100,21 +157,118 @@
...
+				to_chat(user, span_notice("You replace [src]'s mining scanner with a new one."))
+				to_chat(src, span_notice("[user] replaces your current mining scanner with a new one."))
+				scanner.forceMove(get_turf(src))
+			else
+				to_chat(user, span_notice("You insert a mining scanner into [src]."))
+				to_chat(src, span_notice("[user] inserts a new mining scanner into you."))
+			I.forceMove(src)
+			scanner = I
+			return
+		else if(istype(I, /obj/item/pen && user.a_intent == INTENT_HELP))  //So we can get attacked by edaggers
```
Comment (nmajask) [inline code/modules/mining/minebot.dm:244]: Can't do this as you end up stabbing the minebot with the pen

File: code/modules/mining/minebot.dm [hunk 5]
```diff
@@ -259,6 +467,86 @@
 	var/mob/living/simple_animal/hostile/mining_drone/user = owner
 	user.DropOre()
 
+/mob/living/simple_animal/hostile/mining_drone/proc/cycle_weapon()
+	set name = "Cycle Weapon"
+	set desc = "Switch between a plasma cutter or a KA."
+	set category = "Mining Bot"
+
+	if(!cutter)
+		to_chat(usr, span_warning("You don't have a plasma cutter to switch to!"))
```
Comment (nmajask) [inline code/modules/mining/minebot.dm:476]: Should make this prevent you from switching weapons.

File: code/modules/mining/minebot.dm [hunk 6]
```diff
@@ -100,21 +157,118 @@
...
+				to_chat(src, span_notice("[user] replaces your current mining scanner with a new one."))
+				scanner.forceMove(get_turf(src))
+			else
+				to_chat(user, span_notice("You insert a mining scanner into [src]."))
+				to_chat(src, span_notice("[user] inserts a new mining scanner into you."))
+			I.forceMove(src)
+			scanner = I
+			return
+		else if(istype(I, /obj/item/pen && user.a_intent == INTENT_HELP))  //So we can get attacked by edaggers
+			var/new_name = input(user, "Choose a new name for the minig bot", "Name")
```
Comment (nmajask) [inline code/modules/mining/minebot.dm:245]: Sanitize this and run it through the pretty filter, always do that with user input.

**Review by nmajask — CHANGES_REQUESTED** [review]

File: code/modules/mining/minebot.dm (hunk 2 above)
Comment (ghost) [reply code/modules/mining/minebot.dm:187]: ok

**Comment by ghost** [comment]
aaaaaaa

**Comment by Bibby0110** [comment]
stale and doesn't compile. Feel free to reopen when you have less projects in progress :)

**Pull request closed without merging**